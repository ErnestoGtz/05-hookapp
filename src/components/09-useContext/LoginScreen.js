import React,{useState,useContext} from 'react'
import { UserContext } from './UserContext';


export const LoginScreen = () => {
    
    const { setUser } = useContext(UserContext);
    
    return (
        <div>
            <h1>LoginScreen</h1>
            <hr />
            <button 
                className="btn btn-secondary"
                onClick={() => setUser({
                    id:'04131024',
                    name:'Ernesto Gtz'
                })}
            >
                Login
            </button>
        </div>
    )
}
