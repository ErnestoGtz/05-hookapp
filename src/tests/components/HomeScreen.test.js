import React from 'react';
import { mount } from 'enzyme';
import { HomeScreen } from '../../components/09-useContext/HomeScreen';
import '@testing-library/jest-dom';
import { UserContext } from '../../components/09-useContext/UserContext';

describe('Pruebas en <HomeScreen />', () => {
    
    const user = {
        name:'Ernesto',
        email:'ing.ernesto.gutierrez@gmail.com'
    }
    
    const wrapper = mount(
        <UserContext.Provider value={{
            user
        }}>
            <HomeScreen />
        </UserContext.Provider>
        
    );

    test('debe de mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });
    
});

