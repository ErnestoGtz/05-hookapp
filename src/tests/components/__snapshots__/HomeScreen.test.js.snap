// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <HomeScreen /> debe de mostrarse correctamente 1`] = `
<div>
  <h1>
    HomeScreen
  </h1>
  <hr />
  <pre>
    {
   "name": "Ernesto",
   "email": "ing.ernesto.gutierrez@gmail.com"
}
  </pre>
</div>
`;
